lib/Ix/DBIC/Result.pm

```diff
@@ -274,43 +470,250 @@ sub ix_finalize ($class) {
   }
 }
 
+=head1 HOOK METHODS
+
+L<Ix::DBIC::ResultSet> calls out to many methods defined in the rclass during
+normal JMAP method handlers. This allows your rclasses to customize their
+handling in lots of ways, without needing to subclass the ResultSet class
+itself. They come in several varieties:
+
+=begin :list
+
+= check methods (C<ix_FOO_check>)
+
+These are called I<before> the corresponding operation. If they return a value,
+that value must be an L<Ix::Error> object (probably, generated via
+C<< $ctx->error >>). This is especially useful for doing authorization checks.
+
+= error methods (C<ix_FOO_error>)
+
+The methods are passed any error, and expected to return a pair
+C<($row, $error)>. They are useful for modifying the result of an error, or for
+ignoring errors entirely.
```

Comment:
Maybe specify 'return a row to pretend there was no error, or return an error if there was, or return an empty list to let normal error handling take place

Reply:
Yup, have added.